### PR TITLE
Added CustomGlobalExceptionHandler class and validation for existing …

### PR DESCRIPTION
**Pull request by sovisou**
…DTO classes.

### CODE REVIEW

File: src/main/java/com/onlinebookstore/dto/BookDto.java [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookDto {
     private Long id;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/dto/BookDto.java]: Don't need to validate response

File: src/main/java/com/onlinebookstore/dto/BookDto.java [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
     private String isbn;
+    @NotNull
+    @Min(0)
     private BigDecimal price;
+    @NotNull
     private String description;
+    @NotNull
     private String coverImage;
+    @NotNull
     private boolean isDeleted;
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/dto/BookDto.java]: remove isDeleted field

File: src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,14 +1,23 @@
 package com.onlinebookstore.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java]: It's better to use NotBlank for String

```suggestion
    @NotBlank
```

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                                                                  HttpStatusCode status,
+                                                                  WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(e -> getErrorMessage(e))
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java:44]: You can also handle EntityNotFoundException

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: pom.xml [hunk 6]
```diff
@@ -86,6 +86,10 @@
 			<artifactId>liquibase-maven-plugin</artifactId>
 			<version>${liquibase.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.hibernate.validator</groupId>
```
Comment (liliia-ponomarenko) [inline pom.xml]: better use `spring-boot-starter-validation`

File: src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
     private String title;
+    @NotBlank
     private String author;
+    @NotBlank
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java]: You also can validate the size for literal

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> entityNotFoundException(EntityNotFoundException ex,
+                                                            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NOT_FOUND);
+        body.put("message", ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> entityNotFoundException(EntityNotFoundException ex,
+                                                            WebRequest request) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: do you use `request` parameter?

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java:46]: Handle your custom EntityNotFoundException

File: src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java [hunk 11]
```diff
@@ -3,12 +3,12 @@
 import com.onlinebookstore.dto.BookDto;
 import com.onlinebookstore.dto.BookSearchParameters;
 import com.onlinebookstore.dto.CreateBookRequestDto;
-import com.onlinebookstore.exception.EntityNotFoundException;
 import com.onlinebookstore.mapper.BookMapper;
 import com.onlinebookstore.model.Book;
 import com.onlinebookstore.repository.book.BookRepository;
 import com.onlinebookstore.repository.book.BookSpecificationBuilder;
 import com.onlinebookstore.service.BookService;
+import jakarta.persistence.EntityNotFoundException;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java]: Use your custom EntityNotFoundException

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Let's fix issue with EntityNotFoundException;)

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**